csrc/utils.cpp

```diff
@@ -117,6 +117,10 @@ bool is_cpu_scalar(const c10::TensorType& tensor_type) {
       opt_numel.value() == 1;
 }
 
+bool is_meta_scalar(const at::Tensor& tensor) {
+  return tensor.device().is_meta() && tensor.numel() == 1 && tensor.dim() == 0;
```

Comment:
Is it not enough to check `tensor.device().is_meta()`?

Reply:
we do still want to verify that it's a scalar tensor, that's what the validation should do.

Reply:
Oh duh. Of course! Thanks.